### DISCUSSION TITLE
A concept of Trash "location" for removed items

### DISCUSSION THREAD

**Discussion by x80486** [ente-io/ente#492] [Enhancements]
## Description

I would like to see the addition of a `Trash` location for removed items. Currently, when we remove items, they are deleted permanently, which can be problematic if someone accidentally removes an important item.

Having this would offer an added layer of security, ensuring that users can easily recover mistakenly removed items — as it just happened to me :grimacing:

## Describe the Solution You Would Like

A locations where removed items are placed before actually removing them permanently.

## Describe Alternatives You Have Considered

N/A

## Additional Context

Some notes/observations from other applications:

- Users should be able to restore items from the `Trash`
- Items in the `Trash` might be automatically deleted after a certain period to manage storage — not really a pleasant setting, but could be useful for some

**Comment by rareseu** [comment]
Bumping this up. That's a CRITICAL feature that I need. I can't risk the deletion of a token, and it to be simply gone!
I can't 100% switching to Ente Auth

**Reply by 1LuisVargas** [reply]
Agreed. This is a great feature that I'd love to have!

**Comment by vincentcox** [comment]
Yeah this can lock out users of their account PERMANENTLY. Even a 1 day (ideally around 20 days or even a custom setting with minimum of 5min) trash retention can eliminate so much pain. It can also prevent an app error or an attack vector (DoS by locking the user of all his accounts by removing codes).

Additionally, I would ask for the passcode/touchID/... when removing an item. Sometimes the app is left open unlocked on desktop.

**Reply by Damariobros** [reply]
As of the writing of this comment, it already asks for authentication for deleting items when app lock is enabled, we just need the trash location.

**Reply by vincentcox** [reply]
The authentication only prevents an outside attacker, but a self made error not. The trash is really needed, it's also present in other apps for a reason.

**Comment by SkypLabs** [comment]
Seems to be a duplicate of https://github.com/ente-io/ente/discussions/2182.

**Comment by Eternal9100** [comment]
There is a 48 hour delete overun on Authy and this works well.

**Reply by vincentcox** [reply]
exactly! Many new ente users are coming from Authy

**Comment by brog-io** [comment]
This has been implemented with the recent update

**Reply by vincentcox** [reply]
Awesome!

Can confirm:
![F2185DB9-3AC5-4473-8BB3-7ED9493B75E2_1_201_a](https://github.com/user-attachments/assets/f4bdeac8-0138-4ddc-8557-4037ff62b514)

After delete the trash button is there:
<img width="777" alt="image" src="https://github.com/user-attachments/assets/7ff6a2ea-5aa5-4ad0-88b4-3a970b20d307">


Thanks for listening to the community!

**Comment by x80486** [comment]
Do entries in the `Trash` (location) have an expiration date?

I hope they don't...ever :crossed_fingers:

**Reply by ua741** [reply]
There's no expiry for trashed items. The server isn't aware which items are trashed, so we have decided to not add any auto deletion.
